File: ina_2sem/labor_cs/3_lab/1_task/Kolo.java
public class Kolo extends Figura {
    private final double promien;

    public Kolo(double promien) {
        if (promien <= 0) throw new IllegalArgumentException("Promień musi być dodatni.");
        this.promien = promien;
    }

    @Override
    public double obliczPole() {
        return Math.PI * promien * promien;
    }

    @Override
    public double obliczObwod() {
        return 2 * Math.PI * promien;
    }

    @Override
    public String podajNazwe() {
        return "Koło";
    }
}

File: ina_2sem/labor_cs/3_lab/2_task/Kolo.cpp
#include "Kolo.h"

Kolo::Kolo(double r) : promien(r) {
    if (r <= 0) throw std::invalid_argument("Promień musi być dodatni.");
}

double Kolo::obliczPole() {
    return M_PI * promien * promien;
}

double Kolo::obliczObwod() {
    return 2 * M_PI * promien;
}

std::string Kolo::podajNazwe() {
    return "Koło";
}
